library/xml/src/modules/XML.rb

```diff
@@ -165,7 +165,7 @@ def YCPToXMLString(doc_type, contents)
     # @return [Hash] parsed content
     # @raise [XMLDeserializationError] when non supported XML is passed
     def XMLToYCPFile(xml_file)
-      raise XMLDeserializationError, "Cannot find XML file" if SCR.Read(path(".target.size"), xml_file) <= 0
+      raise XMLDeserializationError, "Cannot find XML file" if xml_file.nil? || SCR.Read(path(".target.size"), xml_file) <= 0
```

Comment:
Here I would prefer ArgumentError as for me passing nil is not valid argument and has nothing to do with XMLDeserialization

Reply:
Also it would easier distringuish between not found and nil file...and BTW when you touch it, can you add xml_file path to this exception message?

Reply:
RIght. When dealing with exceptions I find it useful to categorize them like the HTTP status codes: 4xx (app got bad data) or 5xx (app messed up by itself)
Therefore a separate Argument Error (500) seems useful.
And adding the file name is good too.